doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,432 @@
+
+At this point, the zcbor state is initialized.
+Following it, zcbor decoding API calls must be given a pointer to the first zcbor state - `zs` or `&zs[0]` if you prefer, as in the example.
+
+.. note::
+
+   Currently, there is no API provided for zcbor to get information on how much of an input buffer has been consumed for decoding.
+   The only way to find this out is to substitute the buffer pointer, given at zcbor state initialization, from the state object element `payload_mut`, for example `zs->payload_mut - buffer`.
+
+Encoding state initialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Encoding with TinyCBOR is similar to decoding and requires two API calls, one to setup a buffer writer, `cbor_buf_writer_init`, and one to attach the writer to encoding state `cbor_encoder_init`.
+zcbor encoding is initialized with a single call to `zcbor_new_encode_state` that directly ties the provided buffer to encoding state.
+Below is an example of simple zcbor state initialization for encoding:
+
+.. code-block:: c
+
+   zcbor_state_t zs[2];
+   uint8_t buffer[SOME_SIZE];
+   zcbor_new_encode_state(zs, ARRAY_SIZE(zs), buffer, ARRAY_SIZE(buffer), 0);
+   /* Encode data to the buffer with zs state */
+   ...
+
+In the above example, the zcbor state `zs` consists of two states, which is enough when zcbor is in non-canonical configuration (see :kconfig:option:ZCBOR_CANONICAL).
+Otherwise, the number of states would need to be increased by a maximum level of expected lists and arrays.
+The other parameters provided are the `buffer` where data will be placed, with size that limits it, and `0` as the final parameter that represents the number of items encoded into the buffer.
+
+At this point, the zcbor state is initialized and following it zcbor decoding API calls must be given a pointer to the first zcbor state - `zs`, or `&zs[0]`, as in the example.
+
+.. note::
+
+    Currently, there is no API provided for zcbor to get information on how much of an output buffer has been used by encoded data.
+	The only way to find this out is to substitute the buffer pointer, given at a zcbor state initialization, from the state object element `payload_mut`, for example `zs->payload_mut - buffer`.
+
+.. _migration_tinycbor_to_zcbor_general_api:
+
+TinyCBOR to zcbor decoding/encoding API replacement
+---------------------------------------------------
+
+There are three major differences between TinyCBOR and zcbor APIs:
+
+ 1. TinyCBOR calls return error codes while zcbor returns a `bool` value (`true` is success).
+ #. zcbor does not provide type/value peek functions.
+ #. All of zcbor functions advance buffer on success.
+
+zbcor stores error codes with `zs` and, when configured, can automatically fail consecutive calls when decoding or encoding fails.
+Otherwise, the calls are quite similar: functions from both APIs take context/state as one parameter and value or pointer for value and optional buffer size as the following parameters.
```

Comment:
```suggestion
Otherwise, the calls are quite similar: functions from both APIs take context/state as one parameter and value or pointer for value and buffer size, when required, as the following parameters.
```